autorest/azure/token.go

```diff
@@ -301,14 +301,17 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	}
 
 	var newToken Token
+	var aadError AADServiceError
 	err = autorest.Respond(resp,
 		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingJSON(&newToken),
+		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
```

Comment:
oh, i haven't looked at that comment. I am not contesting that - still reviewing. I was answering to @ahmetalpbalkan concern (are you sure this will work?) I will send another iteration for review.